sympy/stats/tests/test_finite_rv.py

```diff
@@ -98,8 +98,8 @@ def test_dice():
 
     assert characteristic_function(X)(t) == exp(6*I*t)/6 + exp(5*I*t)/6 + exp(4*I*t)/6 + exp(3*I*t)/6 + exp(2*I*t)/6 + exp(I*t)/6
     assert moment_generating_function(X)(t) == exp(6*t)/6 + exp(5*t)/6 + exp(4*t)/6 + exp(3*t)/6 + exp(2*t)/6 + exp(t)/6
-    assert median(X) == 3
-    assert median(Y) == 3
+    assert median(X) == FiniteSet(3)
```

Comment:
By returning a set you're suggesting that you're returning the set of all possible medians, but that's not actually the case.
For a regular die 3 is **a** median, but **all** values from Interval(3, 4) can be considered medians. So as a set I would expect the answer to be `Interval(3, 4)`.

I fear I'm repeating myself, but I think that `median` should either return all medians or a single one. Otherwise it's confusing.

Could you please add a precise mathematical definition to the `median` docstring? That makes it easier to have a sensible conversation about what should and should not be returned.

Reply:
The reason is that the domain of Die is FiniteSet(1,2,3,4,5,6)  and therefore only one value 3 is satisfied. Therefore it is not returning an `Interval(3,4)`

Reply:
Please provide a **definition**.
Even if you're restricting the definition to the FiniteSet(1, 2, 3, 4, 5, 6), there's still a second perfectly acceptable median: 4.
At least half of all values are greater than or equal to 4 and
at least half of all values are less than or equal to 4.

Reply:
Median of probablility density function is the Interval or the point where the cumulative density function attains the value of 1/2.

Reply:
> Could you please add a precise mathematical definition to the `median` docstring? That makes it easier to have a sensible conversation about what should and should not be returned.

Yes, i am adding it

Reply:
> Precisely in mathematics, median is the where the following condition satisfy:
> `P(X >= m) >= 1/2` and `P(X <= m) >=1/2`

I know. And you can easily check that this condition is satisfied by `m ==  4`. That is my point, it depends on the precise definition. So the definition you're aiming for is rather 
> Interval or the point where the cumulative density function attains the value of 1/2

I'm still wondering what's to gain by this definition: It still doesn't return all possible medians (see above...), yet it adds the "complication" of wrapping all results in a FiniteSet. (Or is there an example where your current code will actually return an Interval? That would be a good test case to add.)

But as long as the definition you'll add to the docstring is coherent with the test cases it's ok for me.

Reply:
```
P(X <= 4) = P(X==1) + P(X==2) + P(X==3) + P(X==4) = 1/6 + 1/6 +1/6 +1/6 = 2/3
P(X >=4) = P(X==4) + P(X==5) + P(X ==6) = 1/2
```
Therefore 4 is also the median and It should return `FiniteSet(3,4)`. I got that,
I am trying to rectify it. Thanks for pointing it out,
Yes along with corrected implementation, I will also add the mathematical definition in the docstring.